Libraries/Modal/Modal.js

```diff
@@ -44,95 +46,110 @@ import type EmitterSubscription from 'EmitterSubscription';
-     * modal has been shown.
-     *
-     * See https://facebook.github.io/react-native/docs/modal.html#onshow
-     */
-    onShow: PropTypes.func,
-    /**
-     * The `onDismiss` prop allows passing a function that will be called once
-     * the modal has been dismissed.
-     *
-     * See https://facebook.github.io/react-native/docs/modal.html#ondismiss
-     */
-    onDismiss: PropTypes.func,
-    animated: deprecatedPropType(
-      PropTypes.bool,
-      'Use the `animationType` prop instead.',
-    ),
-    /**
-     * The `supportedOrientations` prop allows the modal to be rotated to any of the specified orientations.
-     *
-     * See https://facebook.github.io/react-native/docs/modal.html#supportedorientations
-     */
-    supportedOrientations: PropTypes.arrayOf(
-      PropTypes.oneOf([
-        'portrait',
-        'portrait-upside-down',
-        'landscape',
-        'landscape-left',
-        'landscape-right',
-      ]),
-    ),
-    /**
-     * The `onOrientationChange` callback is called when the orientation changes while the modal is being displayed.
-     *
-     * See https://facebook.github.io/react-native/docs/modal.html#onorientationchange
-     */
-    onOrientationChange: PropTypes.func,
-  };
+type OrientationChangeEvent = SyntheticEvent<
+  $ReadOnly<{|
+    orientation: 'portrait' | 'landscape',
+  |}>,
+>;
+
+type ModalProps = $ReadOnly<{|
+  ...ViewProps,
+
+  /**
+   * The `animationType` prop controls how the modal animates.
+   *
+   * See https://facebook.github.io/react-native/docs/modal.html#animationtype
+   */
+  animationType?: ?('none' | 'slide' | 'fade'),
+
+  /**
+   * The `presentationStyle` prop controls how the modal appears.
+   *
+   * See https://facebook.github.io/react-native/docs/modal.html#presentationstyle
+   */
+  presentationStyle?: ?(
+    | 'fullScreen'
+    | 'pageSheet'
+    | 'formSheet'
+    | 'overFullScreen'
+  ),
+
+  /**
+   * The `transparent` prop determines whether your modal will fill the
+   * entire view.
+   *
+   * See https://facebook.github.io/react-native/docs/modal.html#transparent
+   */
+  transparent?: ?boolean,
+
+  /**
+   * The `hardwareAccelerated` prop controls whether to force hardware
+   * acceleration for the underlying window.
+   *
+   * See https://facebook.github.io/react-native/docs/modal.html#hardwareaccelerated
+   */
+  hardwareAccelerated?: ?boolean,
+
+  /**
+   * The `visible` prop determines whether your modal is visible.
+   *
+   * See https://facebook.github.io/react-native/docs/modal.html#visible
+   */
+  visible?: boolean,
+
+  /**
+   * The `onRequestClose` callback is called when the user taps the hardware
+   * back button on Android or the menu button on Apple TV.
+   *
+   * This is required on Apple TV and Android.
+   *
+   * See https://facebook.github.io/react-native/docs/modal.html#onrequestclose
+   */
+  onRequestClose?: (event?: SyntheticEvent<null>) => mixed,
```

Comment:
This diff fails to land internally because of callsites that pass null to `onRequestClose`. We typically like to make optional arguments nullable to enable spreading props and other things to work. Can you update this type and the others in this diff that are optional to also be nullable?

```
onRequestClose?: ?(event?: SyntheticEvent<null>) => mixed,
```

Reply:
Fixed!